File: api/nodejs/bigquery.js
  // [START bigquery_table_insert_rows]
  // Imports the Google Cloud client library
  const BigQuery = require('@google-cloud/bigquery');

  /**
   * TODO(developer): Uncomment the following lines before running the sample.
   */
  const projectId = "gaming-microservice";
  const datasetId = "demo_dataset";
  const tableId = "gae_performance";
  
  // Creates a client
  const bigquery = new BigQuery({
    projectId: projectId,
  });

function insertBQ(method, elapsed, callback){
  var nowMilliseconds = new Date();
  const rows = [{
  	time: nowMilliseconds,
  	language: "nodejs",
  	method: method,
  	elapsed: elapsed
  }];

//  console.log(rows);
  
  // Inserts data into a table
  bigquery
    .dataset(datasetId)
    .table(tableId)
    .insert(rows)
    .then(() => {
      console.log(`Inserted ${rows.length} rows`);
//      return callback(`Inserted ${rows.length} rows`); //if needed uncomment
    })
    .catch(err => {
      if (err && err.name === 'PartialFailureError') {
      	 //      return callback(err); // if needed uncomment
        if (err.errors && err.errors.length > 0) {
          console.log('Insert errors:');
          err.errors.forEach(err => console.error(err));
          
        }
      } else {
        console.error('ERROR:', err);
 //      return callback(err); // if needed uncomment
      }
    });
}

module.exports = {
	insertBQ : insertBQ
};

File: api/py/bigquery.py
import datetime
import pytz

# https://google-cloud-python.readthedocs.io/en/latest/bigquery/usage.html
from google.cloud import bigquery

def bq_insert_rows(method, elapsed):
	client = bigquery.Client()
	dataset_ref = client.dataset('demo_dataset')
	table_ref = dataset_ref.table('gae_performance')
	table = client.get_table(table_ref)  # API call
	rows_to_insert = [
	    (datetime.datetime.now(pytz.utc), 'python', method, elapsed)
	]
	errors = client.insert_rows(table, rows_to_insert)  # API request
	
	assert errors == []